Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Search.hs

```diff
@@ -131,52 +133,57 @@ handler merchant sReq = do
       merchantId = merchant.id
   result <- getDistanceAndDuration merchantId fromLocationLatLong toLocationLatLong sReq.routeDistance sReq.routeDuration
   logDebug $ "distance: " <> show result.distance
-  mbSpecialLocation <- QSpecialLocation.findSpecialLocationByLatLong fromLocationLatLong
   sessiontoken <- generateGUIDText
   fromLocation <- buildSearchReqLocation merchantId sessiontoken sReq.pickupAddress sReq.customerLanguage sReq.pickupLocation
   toLocation <- buildSearchReqLocation merchantId sessiontoken sReq.dropAddrress sReq.customerLanguage sReq.dropLocation
 
+  mbSpecialLocationTag <- FareProduct.getSpecialLocationTag merchantId fromLocationLatLong toLocationLatLong
+  allFarePolicies <- getFarePolicyForVariants merchantId fromLocationLatLong toLocationLatLong
+  let farePolicies = selectFarePolicy result.distance allFarePolicies
   (quotes, mbEstimateInfos) <-
-    if isJust mbSpecialLocation
+    if isJust mbSpecialLocationTag
       then do
         whenJustM
           (QSearchRequestSpecialZone.findByMsgIdAndBapIdAndBppId sReq.messageId sReq.bapId merchantId)
           (\_ -> throwError $ InvalidRequest "Duplicate Search request")
         searchRequestSpecialZone <- buildSearchRequestSpecialZone sReq merchantId fromLocation toLocation result.distance result.duration
```

Comment:
this is not required in case of `DFareProduct.NORMAL` flowType, right ? if so please move it under `DFareProduct.RIDE_OTP` case match

Reply:
No, use case is that for SpecialLocation...We can have normal estimate ride flow as well. Unlike Jatri Saathi usecase.